grayskull/strategy/py_build.py

```diff
@@ -0,0 +1,144 @@
+            metadata_path = Path(project.metadata_path(output_directory=output))
+
+            distribution = PathDistribution(metadata_path)
+
+            # real distribution name not pathname
+            config.name = distribution.name  # see also _normalized_name
+            config.version = distribution.version
+
+            # grayskull thought the name was the path. correct that.
+            if recipe[0] == '#% set name = "." %}':  # XXX fragile
+                # recipe[0] = x does not work
+                recipe._yaml._yaml_get_pre_comment()[0].value = (
+                    f'#% set name = "{config.name}" %}}\n'
+                    f'#% set version = "{config.version}" %}}\n'
+                )
+
+                recipe["package"]["version"] = "{{ version }}"
+            elif config.name not in str(recipe[0]):
+                log.warning("Package name not found in first line of recipe")
+
+            metadata = distribution.metadata
+
+            requires_python = metadata["requires-python"]
+            if requires_python:
+                requires_python = f"python { requires_python }"
+            else:
+                requires_python = "python"
+
+            recipe["requirements"]["host"] = [requires_python] + sorted(
+                (*build_system_requires, *requires_for_build)
+            )
+
+            requirements = [Requirement(r) for r in distribution.requires]
+            active_requirements = [
+                str(r).rsplit(";", 1)[0]
+                for r in requirements
+                if not r.marker or r.marker.evaluate()
+            ]
+            # XXX to normalize space between name and version, MatchSpec(r).spec
+            normalized_requirements = []
+            for requirement in active_requirements:
+                try:
+                    normalized_requirements.append(
+                        # MatchSpec uses a metaclass hiding its constructor from
+                        # the type checker
+                        MatchSpec(requirement).spec  # type: ignore
+                    )
+                except InvalidMatchSpec:
+                    log.warning("%s is not a valid MatchSpec", requirement)
+                    normalized_requirements.append(requirement)
+
+            # conda does support ~=3.0.0 "compatibility release" matches
+            recipe["requirements"]["run"] = [requires_python] + normalized_requirements
+            # includes extras as markers e.g. ; extra == 'testing'. Evaluate
+            # using Marker().
+
+            recipe["build"]["entry_points"] = [
+                f"{ep.name} = {ep.value}"
+                for ep in distribution.entry_points
+                if ep.group == "console_scripts"
+            ]
+
+            recipe["build"]["noarch"] = "python"
+            recipe["build"][
+                "script"
+            ] = "{{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation"
+            # XXX also --no-index?
```

Comment:
It would be nice to be careful about not fetching pip requirements, even the build-system requirements (poetry, flit-core, hatchling, ...)

Reply:
sorry, I think I didn't follow, what do you mean?